service/history/mutableStateBuilder.go

```diff
@@ -777,7 +777,14 @@ func (e *mutableStateBuilder) GetCronBackoffDuration() time.Duration {
 	if len(info.CronSchedule) == 0 {
 		return backoff.NoBackoff
 	}
-	return backoff.GetBackoffForNextSchedule(info.CronSchedule, e.executionInfo.StartTimestamp, e.timeSource.Now())
+
+	workflowStartEvent, found := e.GetStartEvent()
+	executionTime := e.executionInfo.StartTimestamp
+	if found {
```

Comment:
if this found / not found applies to both initial run of cron workflow & following run of cron workflow, plz include a comment, because normally `e.GetStartEvent()` should be able to load the start event

Reply:
No. Only for continueasnew cron workflow. Not the inital run. So I can just ignore it? When continueasnew, there should be the start event.

Reply:
I updated the comment but keep this found. Just to be safe.

Reply:
no, not exactly

if `e.GetStartEvent()` is for following runs, i.e. calling the existing run to calculate the next run's property, then it must be `found == true`.

so i guess you need to modify this method signature to include a `error`

Reply:
the basic idea is that we should not ignore potential error case

Reply:
Yes. I updated the method to return an error.